Sources/SourceControl/GitRepository.swift

```diff
@@ -391,6 +391,16 @@ public class GitRepository: Repository, WorkingCheckout {
         return localFileSystem.isDirectory(AbsolutePath(firstLine))
     }
 
+    public func getIgnoredFiles() -> [AbsolutePath] {
```

Comment:
This could have a big impact on performance because we will end up looking at each file in the .build directory, which is generally big. We should find a way to do this in a better way.

Reply:
Thanks for the review.
Yeah, this slipped my mind.
Maybe we could use `git check-ignore` on the files found by `findNonSourceFiles`.
Or should we also ignore the files if there isn't a repository for the package?

Reply:
Using `git check-ignore` sounds like a good idea.  And if there isn't a repository for the package, then by definition no files are ignored in a Git sense, so I don't think there's an issue there.

Reply:
ah I meant to say that we should make sure we don't throw when there is no repository.